pkg/operator/operator_config.go

```diff
@@ -558,29 +587,14 @@ func (r *operatorConfigReconciler) ensureRuleEvaluatorDeployment(ctx context.Con
 		return err
 	}
 
-	var projectID, _, _ = resolveLabels(r.opts.ProjectID, r.opts.Location, r.opts.Cluster, spec.ExternalLabels)
-
-	// If no explicit project ID is set, use the one provided to the operator.
-	// On GKE the rule-evaluator can also auto-detect the cluster's project
-	// but this won't work in other Kubernetes environments.
-	queryProjectID := projectID
-	if spec.QueryProjectID != "" {
-		queryProjectID = spec.QueryProjectID
-	}
-	flags := []string{fmt.Sprintf("--query.project-id=%q", queryProjectID)}
+	setContainerExtraArgs(deploy.Spec.Template.Spec.Containers, RuleEvaluatorContainerName, "")
```

Comment:
Note to myself and others: this is killed in other PRs e.g. https://github.com/GoogleCloudPlatform/prometheus-engine/pull/1078